### PR TITLE
Fixes #10853 - add config_template_ids back to organization

### PR DESCRIPTION
**Pull request by mbacovsky**
before_filter converting `config_template_ids` to `provisioning_template_ids` is missing from `taxonomies` which is breaking backward compatibility and hammer bats tests

### CODE REVIEW

**Comment by mbacovsky** [comment]
This is breaking bats tests http://ci.theforeman.org/job/systest_foreman_hammer_nightly/133/default/

File: app/controllers/concerns/api/v2/taxonomies_controller.rb [hunk 1]
```diff
@@ -76,6 +77,13 @@ def destroy
 
   private
 
+  def rename_config_template
+    if params[taxonomy_single.to_sym] && !params[taxonomy_single.to_sym][:config_template_ids].nil?
```
Comment (ares) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb]: Could you please modify `params[taxonomy_single.to_sym]` to `params[taxonomy_single]`? we don't need to convert it to symbol since params are HashWithIndifferentAccess. I know it's not related to this issue but I'd consider the same change everywhere in this controller as acceptable in this PR.

**Comment by ares** [comment]
One more thing, we're missing :provisioning_template_ids in apipie param group.

**Comment by mbacovsky** [comment]
@ares, .to_sym was removed and provisioning_template_ids was added into apipie doc. Note for testing: it propagates to hammer `hammer organization create` but not into `hammer organization add-config-template` which requires hammer patch

**Comment by ares** [comment]
Except one nit it's OK. I created a hammer issue to provide `add-provisioning-template` command at http://projects.theforeman.org/issues/10862. During testing I found and reported one issue http://projects.theforeman.org/issues/10863 related to taxonomies.

**Comment by mbacovsky** [comment]
@ares, `present?` looks better indeed. Thanks for the issues filed

**Comment by ares** [comment]
:+1:  pending tests

**Comment by mbacovsky** [comment]
[test]

File: app/controllers/concerns/api/v2/taxonomies_controller.rb [hunk 2]
```diff
@@ -20,6 +21,7 @@ module Api::V2::TaxonomiesController
       param :compute_resource_ids, Array, N_("Compute resource IDs"), :required => false
       param :media_ids, Array, N_("Media IDs"), :required => false
       param :config_template_ids, Array, N_("Provisioning template IDs"), :required => false
+      param :provisioning_template_ids, Array, N_("Provisioning template IDs"), :required => false
```
Comment (dLobatog) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb:24]: Leave a note - deprecated or something like that

File: app/controllers/concerns/api/v2/taxonomies_controller.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@ module Api::V2::TaxonomiesController
   extend ActiveSupport::Concern
 
   included do
+    before_filter :rename_config_template
```
Comment (dLobatog) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb]: Add `:only` to run it only on the actions where you could be passing the parameter. Also it looks like you can probably avoid going in the method with an `:if` here `if params[taxonomy_single] && params[taxonomy_single][:config_template_ids].present?`

File: app/controllers/concerns/api/v2/taxonomies_controller.rb [hunk 4]
```diff
@@ -76,12 +78,19 @@ def destroy
...
   def params_match_database
     # change params[:select_all_types] to params[:ignore_types] to match database
-    if params[taxonomy_single.to_sym].try(:[], :select_all_types)
-      params[taxonomy_single.to_sym][:ignore_types] = params[taxonomy_single.to_sym][:select_all_types]
-      params[taxonomy_single.to_sym]                = params[taxonomy_single.to_sym].reject { |k, v| k == "select_all_types" }
-      return params[taxonomy_single.to_sym]
+    if params[taxonomy_single].try(:[], :select_all_types)
+      params[taxonomy_single][:ignore_types] = params[taxonomy_single][:select_all_types]
+      params[taxonomy_single]                = params[taxonomy_single].reject { |k, v| k == "select_all_types" }
+      return params[taxonomy_single]
```
Comment (dLobatog) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb:93]: Does this return serves any purpose? Maybe you can do an early `return if params[taxonomy_single].try(:[], :select_all_types)` and avoid nesting the whole method in one if. The way it is right now it looks like you could avoid calling this method by just adding something like `:if => { params[taxonomy_single].try(:[], :select_all_types) }`on the filter

Comment (ares) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb:93]: this was originally not touched by this PR, @mbacovsky just removed `.to_sym` in this method

File: app/controllers/concerns/api/v2/taxonomies_controller.rb (hunk 2 above)
Comment (ares) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb:24]: I was against custom deprecated string here, it's planned to have support for marking params as deprecated in apipie, maybe leaving a comment to easily find this later would be enough

File: app/controllers/concerns/api/v2/taxonomies_controller.rb (hunk 3 above)
Comment (mbacovsky) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb]: @eLobato, :only added

File: app/controllers/concerns/api/v2/taxonomies_controller.rb (hunk 2 above)
Comment (mbacovsky) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb:24]: @eLobato I've added just a FIXME comment to locate the place. We should add support for deprecated options into Apipie first. Then this is information could be reused in docs and Hammer. I filed issue as a reminder  http://projects.theforeman.org/issues/10869

File: app/controllers/concerns/api/v2/taxonomies_controller.rb (hunk 4 above)
Comment (mbacovsky) [inline app/controllers/concerns/api/v2/taxonomies_controller.rb:93]: @eLobato as @ares suggests I'd prefer to touch the unrelated stuff as little as possible.

**Comment by ares** [comment]
@eLobato anything that prevents this PR to be merged or can I go ahead?

**Comment by dLobatog** [comment]
Merged as ebceac9ae390686eb4cca86b3cf815f2d3312f57, thanks @mbacovsky!
